Automattic/sensei
# Add course completion rate to usage tracking data

### Changes proposed in this Pull Request

Adds a `course_completion_rate` event property as part of the `sensei_stats_log` event. The rate is calculated by getting all published courses, and then for each course:

1. Determining the number of non-admin learners who are enrolled in the course.
2. Determining the number of non-admin learners who are enrolled in and have completed the course.
3. Calculating the completion rate for the course by dividing the result of step 2 by the result of step 1.
4. Calculating the average course completion rate of all courses.

Instead of using enrolment data, I pondered calculating the rate by using the number of learners who started the course vs. the number of learners who completed it. I decided against this since ultimately we want to make changes to the product to improve this rate, but we won't be able to move the needle as much if we include learners who are not enrolled in the first place.

### Testing instructions

* Enrol some admin and non-admin learners in some courses.
* Complete the course for some of the users in some of the courses.
* Do some math to calculate what the course completion rate should be.
* Enable usage tracking.
* Using WP Crontrol, run the `sensei_usage_tracking_send_usage_data` job.
* Ensure that the value of the `course_completion_rate` property for the `sensei_stats_log` event matches the rate you calculated.
